spec.bs

```diff
@@ -4372,319 +4380,296 @@ value is used to denote that the currency is unspecified.
-  1. Execute the steps for error-handling.
-  1. Decrement |auctionConfig|'s [=auction config/pending promise count=].
 
+  1. Increment |auctionConfig|'s [=auction config/pending promise count=].
+  1. Let |resolvedAndTypeChecked| be the promise representing performing the following steps
+    [=upon fulfillment=] of |p| with |result|:
+    1. Execute the steps to be run for parsing of the value given |result|.
+    1. If no exception was [=exception/thrown=] in the previous step, then decrement
+      |auctionConfig|'s [=auction config/pending promise count=].
+  1. [=Upon rejection=] of |resolvedAndTypeChecked|:
+    1. Execute the steps for error-handling.
+    1. Decrement |auctionConfig|'s [=auction config/pending promise count=].
 </div>
 
 <div algorithm>
-  To <dfn>look up per-buyer currency</dfn> given an [=auction config=] |auctionConfig|, and an [=origin=] |buyer|:
+To <dfn>look up per-buyer currency</dfn> given an [=auction config=] |auctionConfig|, and an
+[=origin=] |buyer|:
 
   1. Let |perBuyerCurrency| be |auctionConfig|'s [=auction config/all buyers currency=]
-  1. Assert: |auctionConfig|'s [=auction config/per buyer currencies=] is an [=ordered map=].
+  1. [=Assert=] that |auctionConfig|'s [=auction config/per buyer currencies=] is an [=ordered map=].
   1. If |auctionConfig|'s [=auction config/per buyer currencies=][|buyer|] [=map/exists=], then set
     |perBuyerCurrency| to |auctionConfig|'s [=auction config/per buyer currencies=][|buyer|].
   1. Return |perBuyerCurrency|
 </div>
 
-<h3 dfn-type=dfn>Per buyer bid generator</h3>
+<h3 id=bid-generators>Bid generator</h3>
 
-A per buyer bid generator is an [=ordered map=] whose [=map/keys=] are [=URLs=] representing
-[=interest group/trusted bidding signals urls=], and whose [=map/values=] are
+A <dfn>per buyer bid generator</dfn> is an [=ordered map=] whose [=map/keys=] are [=URLs=]
+representing [=interest group/trusted bidding signals urls=], and whose [=map/values=] are
 [=per signals url bid generators=].
 
-<h3 dfn-type=dfn>Per signals url bid generator</h3>
-
-A per signals url bid generator is an [=ordered map=] whose [=map/keys=] are [=origins=]
+A <dfn>per signals url bid generator</dfn> is an [=ordered map=] whose [=map/keys=] are [=origins=]
 representing [=interest group/joining origins=], and whose [=map/values=] are [=lists=] of
 [=interest groups=].
 
+<h3 id=generated-bid-header>Generated bid</h3>
 
-<h3 dfn-type=dfn>Previous win</h3>
-
-The [=interest group=]'s auction win history, to allow on-device frequency capping.
-
-
-<dl dfn-for="previous win">
-: <dfn>time</dfn>
-:: A [=moment=]. Approximate time the [=interest group=] won an auction.
-: <dfn>ad json</dfn>
-:: A [=string=]. A JSON serialized object corresponding to the ad that won the auction.
+A <dfn>generated bid</dfn> is a bid that needs to be scored by the seller. The bid is either the
+output of running a Protected Audience `generateBid()` script, or an [=additional bid=] provided by

```

Comment:
Can we link to something more specific for `generateBid()`? Perhaps the section describing what that is and how it's implemented?